definitions/npm/jasmine_v2.4.x/flow_v0.25.x-/jasmine_v2.4.x.js

```diff
@@ -62,11 +62,13 @@ type JasmineSpyStrategyType = {
   throwError(errorMessage?: string): JasmineSpyType
 };
 
-type JasmineSpyType = {
+type JasmineSpyTypeProto = {
   and: JasmineSpyStrategyType,
   calls: JasmineCallsType
 };
 
+type JasmineSpyType = JasmineSpyTypeProto & Function;
```

Comment:
Ahh TIL, didnt realise they they warn off using Function.
I have tried a few other approaches using mixed and generics but I cant seem to get anything else to work.

Any suggestions?

Reply:
Isnt `() => any` just as bad as using `Function`?

Reply:
@AndrewSouthpaw looking at the definition of jasmine it already uses `Function` a lot. 
I believe it is because it needs to create a very generic function which may or may not return anything. I don't think there is a better way to do this.

Reply:
Works for me.